src/OpenTelemetry/Internal/SelfDiagnosticsConfigRefresher.cs

```diff
@@ -202,8 +202,26 @@ private void OpenLogFile(string newLogDirectory, int newFileSize)
             try
             {
                 Directory.CreateDirectory(newLogDirectory);
-                var fileName = Path.GetFileName(Process.GetCurrentProcess().MainModule.FileName) + "."
-                    + Process.GetCurrentProcess().Id + ".log";
+
+                var currentProcess = Process.GetCurrentProcess();
+
+                string machineName;
+
+                try
+                {
+                    machineName = Environment.MachineName;
+                }
+                catch
+                {
+                    machineName = "UnknownMachineName";
+                }
+
+                var fileName = DateTime.UtcNow.ToString("yyyyMMdd'T'HHmmss'Z'")
+                    + '.' + machineName
+                    + '.' + Path.GetFileName(currentProcess.MainModule.FileName)
```

Comment:
How would the consumer know which part is the machine name vs. module/file name?